src/viking.js

```diff
@@ -1,11 +1,103 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+    attack() {
+        return this.strength
+    }
+    receiveDamage(damage) {
+        this.health -= damage;
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage; 
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`;
+        }
+        if (this.health <= 0) {
+            return `${this.name} has died in act of combat`;
+        }
+    }
+
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
+
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength)
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`;
+        }
+        if (this.health <= 0) {
+            return "A Saxon has died in combat";
+        }
+    }
+}
+    
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [];
+        this.saxonArmy = [];
+    }
+
+    addViking(Viking) {
+        this.vikingArmy.push(Viking);
+    }
+
+    addSaxon(Saxon) {
+        this.saxonArmy.push(Saxon);
+    }
+
+    vikingAttack() {
+        const randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+        const randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+        const saxonDamage = randomSaxon.receiveDamage(randomViking.strength);
+        this.saxonArmy = this.saxonArmy.filter(Saxon => Saxon.health > 0);
+        return saxonDamage;
+     }
+
+     saxonAttack() {
+         const randomViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+         const randomSaxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+         const vikingDamage = randomViking.receiveDamage(randomSaxon.strength);
+         this.vikingArmy = this.vikingArmy.filter(Viking => Viking.health > 0);
+         return vikingDamage
+     }
+
+     showStatus() {
+         if (this.saxonArmy.length === 0) {
+             return "Vikings have won the war of the century!" }
+             else if (this.vikingArmy.length === 0) {
+                 return "Saxons have fought for their lives and survived another day..." }
+                 else if (this.saxonArmy.length === this.vikingArmy.length) {
```

Comment:
That's incorrect, the the right condition for Vikings and Saxons to still be at war is for their armies to exist
```
else if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0) {
      return 'Vikings and Saxons are still in the thick of battle.';
    }